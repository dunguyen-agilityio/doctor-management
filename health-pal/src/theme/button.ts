import { Button as TamaguiButton, styled } from 'tamagui'

export const Button = styled(TamaguiButton, {
  fontFamily: '$body',
  fontWeight: '500',
  fontSize: '$s',
  disabledStyle: { opacity: 0.75, backgroundColor: '$grey900' },
  variants: {
    variant: {
      primary: {
        backgroundColor: '$primary',
        color: '$white',
        borderRadius: '$6',
        height: 48,
        paddingVertical: 12,
        paddingHorizontal: 20,
        pressStyle: { backgroundColor: '$primary', opacity: 0.85 },
        hoverStyle: { opacity: 0.85 },
      },
      secondary: {
        backgroundColor: '$grey200',
        color: '$primary',
        borderRadius: '$6',
        height: 48,
        paddingVertical: 12,
        paddingHorizontal: 20,
        pressStyle: { backgroundColor: '$grey200' },
        hoverStyle: { opacity: 0.85 },
      },
      outlined: {
        color: '$primary',
        backgroundColor: 'transparent',
        py: 10,
        borderWidth: 1,
        borderColor: '$grey200',
        borderRadius: '$1',
        height: 40,
      },
      icon: {
        width: 24,
        height: 24,
        borderRadius: 24,
      },
    },
    full: {
      true: { minWidth: '100%' },
    },
  } as const,
  defaultVariants: { variant: 'primary' },
})
