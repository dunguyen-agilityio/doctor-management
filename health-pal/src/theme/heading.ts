import { styled, Text as TamaguiText } from 'tamagui'

// Text Component with Variants
export const Heading = styled(TamaguiText, {
  fontFamily: '$heading',
  fontWeight: '700',
  color: '$primary',
  variants: {
    size: {
      small: {
        fontSize: '$s',
        lineHeight: '$s',
      },
      medium: {
        fontSize: '$md',
        lineHeight: '$md',
      },
      large: {
        fontSize: '$lg',
        lineHeight: '$lg',
      },
      extraLarge: {
        fontSize: '$xl',
        lineHeight: '$xl',
      },
    },
  } as const,
  defaultVariants: { size: 'medium' },
})
