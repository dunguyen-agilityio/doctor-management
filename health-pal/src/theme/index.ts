export * from './button'
export * from './text'
export * from './heading'
export * from './input'
export * from './stack'
