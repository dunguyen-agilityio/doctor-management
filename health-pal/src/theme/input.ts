import { styled, Input as TamaguiInput } from 'tamagui'

export const Input = styled(TamaguiInput, {
  fontFamily: '$body',
  fontWeight: '400',
  fontSize: '$s',
  variants: {
    variant: {
      flat: {
        backgroundColor: '$grey50',
        color: '$grey600',
        borderRadius: '$1',
        height: 45,
        paddingVertical: 12,
        paddingHorizontal: 16,
        borderColor: '$grey300',
        placeholderTextColor: '$grey400',
      },
      bordered: {
        color: '$primary',
        backgroundColor: 'transparent',
        py: 10,
        borderWidth: 1,
        borderColor: '$grey200',
        borderRadius: '$1',
        height: 40,
      },
    },
  } as const,
  defaultVariants: { variant: 'flat' },
})
