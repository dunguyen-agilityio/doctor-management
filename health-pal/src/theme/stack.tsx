import { XStack as TamaguiXStack, YStack as TamaguiYStack, styled } from 'tamagui'

export const YStack = styled(TamaguiYStack, {
  backgroundColor: 'white',
  flex: 1,
})

export const XStack = styled(TamaguiXStack, {
  backgroundColor: 'white',
})
