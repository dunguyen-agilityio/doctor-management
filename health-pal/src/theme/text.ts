import { styled, Text as TamaguiText } from 'tamagui'

// Text Component with Variants
export const Text = styled(TamaguiText, {
  fontFamily: '$body',
  fontWeight: 400,
  variants: {
    size: {
      extraSmall: {
        fontSize: '$xs',
        lineHeight: '$xs',
      },
      small: {
        fontSize: '$s',
        lineHeight: '$s',
      },
      medium: {
        fontSize: '$md',
        lineHeight: '$md',
      },
      large: {
        fontSize: '$lg',
        lineHeight: '$lg',
      },
      extraLarge: {
        fontSize: '$xl',
        lineHeight: '$xl',
      },
    },
    color: {
      primary: { color: '$primary' },
      secondary: { color: '$white' },
      muted: { color: '$blue' },
    },
  } as const,
  defaultVariants: { size: 'medium', color: 'primary' },
})
