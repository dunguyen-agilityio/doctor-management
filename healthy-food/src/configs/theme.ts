export const COLOR = {
  PRIMARY: '#5c5c5c',
  WHITE: '#fff',
  LIGHT_GREEN: '#91c788',
  LIGHT_GRAY: '#ececec',
  GRAY: '#7e7e7e',
  RED: '#ea1502',
  PURPLE: '#9059a7',
  BLACK: '#000000',
  LIGHT_BLACK: '#330600',
  WARNING: '#da6317',
  LIGHT_PURPLE: '#fff3f1',
  GREEN: '#27ae60',
  ORANGE: '#d89e4a',
  YELLOW: '#f2c94c',
  LIGHT_GREEN_10: 'rgba(28, 195, 121, 0.1)',
  DARK_GREEN: '#1cc379',
  SECONDARY: '#ff8473',
  BLACK_BEAN: '#330600',
  GREY: '#e0e0e0',
  TERTIARY: '#696969',
  LIGHT_GREY: '#dbdbdb',
};

export enum TEXT_COLOR {
  PRIMARY = 'PRIMARY',
  SECONDARY = 'SECONDARY',
  TERTIARY = 'TERTIARY',
  FOURTH = 'FOURTH',
}
