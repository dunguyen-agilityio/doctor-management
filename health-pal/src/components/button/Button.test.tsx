import { render } from '@utils-test'

import { Button } from '@app/theme'

describe('MyButton', () => {
  test('Text renders correctly on MyButton', () => {
    const { getByText, toJSON } = render(<Button>My Button!</Button>)
    const tree = toJSON()
    expect(tree).toMatchSnapshot()

    expect(getByText('My Button!')).toBeVisible()
  })
})
