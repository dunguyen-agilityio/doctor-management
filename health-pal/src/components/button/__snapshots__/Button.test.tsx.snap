// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton Text renders correctly on MyButton 1`] = `
<View
  focusVisibleStyle={{}}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="button"
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1c2a3a",
      "borderBottomColor": "transparent",
      "borderBottomLeftRadius": 60,
      "borderBottomRightRadius": 60,
      "borderBottomWidth": 1,
      "borderLeftColor": "transparent",
      "borderLeftWidth": 1,
      "borderRightColor": "transparent",
      "borderRightWidth": 1,
      "borderStyle": "solid",
      "borderTopColor": "transparent",
      "borderTopLeftRadius": 60,
      "borderTopRightRadius": 60,
      "borderTopWidth": 1,
      "color": "#ffffff",
      "flexDirection": "row",
      "flexWrap": "nowrap",
      "fontWeight": "500",
      "height": 48,
      "justifyContent": "center",
      "paddingBottom": 12,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 12,
    }
  }
>
  <Text
    lineBreakMode="clip"
    numberOfLines={1}
    style={
      {
        "color": "#ffffff",
        "flexGrow": 0,
        "flexShrink": 1,
        "fontFamily": "Inter_500Medium",
        "fontSize": 16,
        "lineHeight": 24,
        "userSelect": "none",
      }
    }
    suppressHighlighting={true}
  >
    My Button!
  </Text>
</View>
`;
